src/service.rs

```diff
@@ -477,16 +485,19 @@ impl ServiceStatus {
         raw_status
     }
 
-    fn from_raw(raw_status: winsvc::SERVICE_STATUS) -> Result<Self> {
+    /// Tries to parse a `SERVICE_STATUS` into a Rust [`ServiceStatus`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the `dwCurrentState` field does not represent a valid [`ServiceState`].
+    fn from_raw(raw: winsvc::SERVICE_STATUS) -> Result<Self, TryFromIntError> {
```

Comment:
I don't think `winsvc::SERVICE_STATUS` is an integer. 😄 You don't have to follow the std, you can simply Add `TryFromRawError`

Reply:
Good input! I completely forgot that not all these errors are from parsing integers.. Renamed it a bit.